packages/dapp/src/components/Dashboard/ClaimRewardsItem/ClaimRewardsItemReduxWrapper.tsx

```diff
@@ -0,0 +1,165 @@
+import * as React from "react";
+import { connect, DispatchProp } from "react-redux";
+import { BigNumber } from "bignumber.js";
+
+import { State } from "../../../redux/reducers";
+import {
+  makeGetListingAddressByChallengeID,
+  makeGetListingAddressByAppealChallengeID,
+  makeGetUserChallengeData,
+  makeGetUserAppealChallengeData,
+  makeGetChallenge,
+  makeGetParameterProposalChallenge,
+  getAppealChallengeParentChallenge,
+  makeGetUserProposalChallengeData,
+  makeGetProposalByChallengeID,
+} from "../../../selectors";
+import { getContent } from "../../../redux/actionCreators/newsrooms";
+import { fetchAndAddChallengeData } from "../../../redux/actionCreators/challenges";
+import { fetchAndAddParameterProposalChallengeData } from "../../../redux/actionCreators/parameterizer";
+import { ClaimRewardsItemOwnProps, ClaimRewardsViewComponentProps, ProposalClaimRewardsComponentProps } from "./types";
+import { ClaimRewardsViewComponent, ProposalClaimRewardsViewComponent } from "./ClaimRewardsViewComponents";
+
+interface ClaimRewardsChallengeProp {
+  challenge?: any;
+  challengeDataRequestStatus?: any;
+}
+
+class ClaimRewardsItemReduxWrapperComponent extends React.Component<
+  ClaimRewardsItemOwnProps & ClaimRewardsViewComponentProps & ClaimRewardsChallengeProp & DispatchProp<any>
+> {
+  public async componentDidMount(): Promise<void> {
+    await this.ensureListingAndNewsroomData();
+  }
+
+  public async componentDidUpdate(): Promise<void> {
+    await this.ensureListingAndNewsroomData();
+  }
+
+  public render(): JSX.Element {
+    return <ClaimRewardsViewComponent {...this.props} />;
+  }
+
+  private ensureListingAndNewsroomData = async (): Promise<void> => {
+    const { newsroom, challengeID, challenge, challengeDataRequestStatus, dispatch } = this.props;
+    if (newsroom && newsroom.data && newsroom.data.charterHeader) {
+      dispatch!(await getContent(newsroom.data.charterHeader));
+    }
+    if (challengeID && !challenge && !challengeDataRequestStatus) {
+      dispatch!(fetchAndAddChallengeData(challengeID! as string));
+    }
+  };
+}
+
+class ClaimRewardsProposalItemReduxWrapperComponent extends React.Component<
+  ClaimRewardsItemOwnProps & ProposalClaimRewardsComponentProps & ClaimRewardsChallengeProp & DispatchProp<any>
+> {
+  public async componentDidMount(): Promise<void> {
+    await this.ensureProposalData();
+  }
+
+  public async componentDidUpdate(): Promise<void> {
+    await this.ensureProposalData();
```

Comment:
should this be only if relevant props have changed?

Reply:
`ensureProposalData` does some checking before it dispatches the action to fetch this data.